mithril-common/src/store/adapter/sqlite_adapter.rs

```diff
@@ -0,0 +1,500 @@
+        let test_name = "test_store_record";
+        let mut adapter = init_db(test_name);
+        adapter
+            .store_record(&1, "one".to_string().borrow())
+            .await
+            .unwrap();
+        let filepath = get_file_path(test_name);
+        let connection = Connection::open(&filepath).expect(&format!(
+            "Expecting to be able to open SQLite file '{}'.",
+            filepath.to_string_lossy()
+        ));
+        let mut statement = connection
+            .prepare(format!("select key_hash, key, value from {}", TABLE_NAME))
+            .unwrap()
+            .into_cursor();
+        let row = statement
+            .try_next()
+            .unwrap()
+            .expect("Expecting at least one row in the result set.");
+        assert_eq!(
+            1,
+            row[1]
+                .as_integer()
+                .expect("expecting field 1 to be an integer")
+        );
+        assert_eq!(
+            "\"one\"".to_string(),
+            row[2]
+                .as_string()
+                .expect("expecting field 2 to be a string")
+        );
+    }
+
+    #[tokio::test]
+    async fn test_get_record() {
+        let test_name = "test_get_record";
+        let mut adapter = init_db(test_name);
+        adapter
+            .store_record(&1, "one".to_string().borrow())
+            .await
+            .unwrap();
+        adapter
+            .store_record(&2, "two".to_string().borrow())
+            .await
+            .unwrap();
+        adapter
+            .store_record(&3, "three".to_string().borrow())
+            .await
+            .unwrap();
+        assert_eq!(
+            Some("one".to_string()),
+            adapter.get_record(&1).await.unwrap()
+        );
+        assert_eq!(
+            Some("three".to_string()),
+            adapter.get_record(&3).await.unwrap()
+        );
+        assert_eq!(
+            Some("two".to_string()),
+            adapter.get_record(&2).await.unwrap()
+        );
+        assert_eq!(None, adapter.get_record(&4).await.unwrap());
+    }
+
+    #[tokio::test]
+    async fn test_get_iterator() {
+        let test_name = "test_get_iterator";
+        let mut adapter = init_db(test_name);
+        adapter
+            .store_record(&1, "one".to_string().borrow())
+            .await
+            .unwrap();
+        adapter
+            .store_record(&2, "two".to_string().borrow())
+            .await
+            .unwrap();
+        adapter
+            .store_record(&3, "three".to_string().borrow())
+            .await
+            .unwrap();
+        let iterator = adapter.get_iter().await.unwrap();
+
+        for (index, element) in iterator.enumerate() {
+            match index {
+                0 => assert_eq!("one", element),
+                1 => assert_eq!("two", element),
+                2 => assert_eq!("three", element),
+                i => panic!("unexpected result index {} with data = '{:?}'.", i, element),
+            }
```

Comment:
Why don't you simply collect the iterator and compare it's result with a unique `assert_eq` ?

Reply:
I found it more convenient that way to add/remove test cases.

Reply:
I found it more readable and as convenient to add new cases:
```rust
assert_eq!(
    vec![
        (0, "one".to_string()),
        (1, "two".to_string()),
        (2, "three".to_string()),
    ],
    iterator.enumerate().collect::<Vec<(usize, String)>>(),
);
```
Btw: the test is wrong, the iterator should returns elements in the reverse order (from the latest to the oldest).

Reply:
An alternative using the iterator would be:
```rust
let mut iterator = adapter.get_iter().await.unwrap().enumerate();

assert_eq!(Some((2_usize, "three".to_string())), iterator.next());
assert_eq!(Some((1_usize, "two".to_string())), iterator.next());
assert_eq!(Some((0_usize, "one".to_string())), iterator.next());
assert_eq!(None, iterator.next());
```